.github/workflows/regression.yml

```diff
@@ -0,0 +1,67 @@
+name: regression
+
+on:
+  push:
+    branches: [main]  # Adjust this if your default branch has a different name
+  pull_request:
+    types: [labeled]
+  workflow_dispatch:  # Add manual trigger capability
+
+jobs:
+  regression:
+    runs-on: blacksmith-4vcpu-ubuntu-2204
```

Comment:
Went fairly small on the CPUs here because I think it's likely we have one or two jobs taking waayyyy longer than the others